File: PROBLEMS/TWO-SUM/soln3.ts
const arr:number[]=[4, 5, 1, 7, 2];
const k:number=3;

const twoSum =(arr:number[],k:number):boolean=>{
    const set:Set<number>=new Set();
    for(let i:number=0;i<arr.length;i++){
     const diff:number=k-arr[i];
     if(set.has(diff)){
        return true
     }else{
        set.add(arr[i])
     }
    }
    return false;
}

console.log(twoSum(arr,k))
//O(n)

File: soln3.rb
require 'set'

arr = [4, 5, 1, 7, 2]
k = 3

def two_sum(arr, k)
  my_set = Set.new

  for i in 0...arr.length
    diff = k - arr[i]
    if my_set.include?(diff)
      return true
    else
      my_set.add(arr[i])
    end
  end

  return false
end

puts two_sum(arr, k).to_s
